code/search/indexes/SearchIndex.php

```diff
@@ -336,14 +356,17 @@ public function addAllFulltextFields($includeSubclasses = true)
     {
         foreach ($this->getClasses() as $class => $options) {
             foreach (SearchIntrospection::hierarchy($class, $includeSubclasses, true) as $dataclass) {
-                $fields = DataObject::database_fields($dataclass);
-
+                $fields = DataObject::getSchema()->databaseFields($class);
                 foreach ($fields as $field => $type) {
                     if (preg_match('/^(\w+)\(/', $type, $match)) {
                         $type = $match[1];
                     }
-                    list($type, $args) = Object::parse_class_spec($type);
-                    if (is_subclass_of($type, 'StringField')) {
+                    list($type, $args) = ClassInfo::parse_class_spec($type);
+
+                    // Get class from shortName
+                    $object = Injector::inst()->get($type, false, ['Name' => 'test']);
+
+                    if (is_subclass_of(get_class($object), 'SilverStripe\ORM\FieldType\DBString')) {
```

Comment:
This would be better as `if ($object instanceof DBString) {` with a namespace import at the top